hedera-node/src/main/java/com/hedera/services/fees/calculation/consensus/txns/UpdateTopicResourceUsage.java

```diff
@@ -46,33 +46,36 @@ public boolean applicableTo(TransactionBody txn) {
     }
 
     @Override
-    public FeeData usageGiven(TransactionBody txn, SigValueObj sigUsage, StateView view) throws InvalidTxBodyException {
-        try {
-            MerkleTopic merkleTopic = view.topics().get(MerkleEntityId.fromTopicId(txn.getConsensusUpdateTopic().getTopicID()));
-            long rbsIncrease = getUpdateTopicRbsIncrease(
-                    txn.getTransactionID().getTransactionValidStart(),
-                    JKey.mapJKey(merkleTopic.getAdminKey()),
-                    JKey.mapJKey(merkleTopic.getSubmitKey()),
-                    merkleTopic.getMemo(),
-                    merkleTopic.hasAutoRenewAccountId(),
-                    lookupExpiry(merkleTopic),
-                    txn.getConsensusUpdateTopic());
-            return getConsensusUpdateTopicFee(txn, rbsIncrease, sigUsage);
-        } catch (Exception illegal) {
-            log.warn("Usage estimation unexpectedly failed for {}!", txn, illegal);
-            throw new InvalidTxBodyException(illegal);
+    public FeeData usageGiven(final TransactionBody txnBody, final SigValueObj sigUsage, final StateView view) throws InvalidTxBodyException {
+        if (txnBody == null || !txnBody.hasConsensusUpdateTopic()) {
+            throw new InvalidTxBodyException("consensusUpdateTopic field not available for Fee Calculation");
         }
-    }
+        if (view == null) {
+            throw new IllegalStateException("No StateView present !!");
+        }
+
+        long rbsIncrease = 0;
+        MerkleTopic merkleTopic = view.topics().get(
+                MerkleEntityId.fromTopicId(txnBody.getConsensusUpdateTopic().getTopicID()));
 
-    private Timestamp lookupExpiry(MerkleTopic merkleTopic) {
-        try {
-            return Timestamp.newBuilder()
-                    .setSeconds(merkleTopic.getExpirationTimestamp().getSeconds())
-                    .build();
-        } catch (NullPointerException e) {
-			/* Effect is to charge nothing for RBH */
-            log.warn("Missing topic expiry data, ignoring expiration time in fee calculation!", e);
-            return ZERO_EXPIRY;
+        if (merkleTopic != null && merkleTopic.hasAdminKey()) {
```

Comment:
We should split this logic out.
What should we do on `null` merkleTopic (not found)? I fail to see how, with this new code, we pass the newly added EET for the `INVALID_TOPIC_ID` scenario.

Reply:
Both scenarios, 
1. missing topic
2. immutable topic

should result in a fee calculation with `rbsIncrease = 0;`

This code avoids a NPE log. I added that spec as we are not covering that scenario in TopicUpdateSuite

Reply:
@anighanta, here is my concern. When `merkleTopic == null`, the topic is not found in the current state view. Our code has to return an `INVALID_TOPIC_ID` response. Without a throw for this scenario, I don't see how we escalate back to to the stack trace to `handleTransaction` to return an `INVALID_TOPIC_ID` response code. We do need a test for this scenario and make sure it pass, because otherwise we already changed the logic and the response to the client.

Reply:
we handle that here https://github.com/hashgraph/hedera-services/blob/46981dd6fe83e0e0228ba2e626f3264b3ccb8d28/hedera-node/src/main/java/com/hedera/services/txns/validation/ContextOptionValidator.java#L149

Reply:
Great! Thanks!

Reply:
@qnswirlds yes @anighanta's fix here is just to make sure `UpdateTopicResourceUsage` doesn't hit a NPE while doing its best to estimate the resource usage of the `UpdateTopic` txn.

The final status for the txn will be set later by the `TransitionLogic`; and if the topic is missing, that will be `INVALID_TOPIC_ID`.